pkg/kubectl/run.go

```diff
@@ -282,6 +302,94 @@ func (JobV1Beta1) Generate(genericParams map[string]interface{}) (runtime.Object
 	return &job, nil
 }
 
+type JobV1 struct{}
+
+func (JobV1) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"labels", false},
+		{"default-name", false},
+		{"name", true},
+		{"image", true},
+		{"port", false},
+		{"hostport", false},
+		{"stdin", false},
+		{"leave-stdin-open", false},
+		{"tty", false},
+		{"command", false},
+		{"args", false},
+		{"env", false},
+		{"requests", false},
+		{"limits", false},
+		{"restart", false},
+	}
+}
+
+func (JobV1) Generate(genericParams map[string]interface{}) (runtime.Object, error) {
+	args, err := getArgs(genericParams)
+	if err != nil {
+		return nil, err
+	}
+
+	envs, err := getV1Envs(genericParams)
```

Comment:
Since we're targeting to get rid of internal representations entirely from kubectl (as pointed in #3955). I've decided to go with the code duplication for now. I'm planning to create a followup, probably next week, converting all to versioned objects. I'm aware it won't make it 1.2 but still I thought about it as the easiest option.